libbuteosyncfw/pluginmgr/plugin_main.cpp

```diff
@@ -75,23 +76,19 @@ int main( int argc, char** argv )
                        << profileName << " registered at dbus "
                        << DBUS_SERVICE_NAME_PREFIX + profileName
                        << " and path " << DBUS_SERVICE_OBJ_PATH );
+            // TODO: Should any unix signals be handled?
+            retn = app.exec();
+            connection.unregisterObject(DBUS_SERVICE_OBJ_PATH);
         } else {
-            QByteArray errorMsg = QString(QLatin1String(
-                    "Unable to register dbus object for service: %1"))
-                    .arg(servicePath).toLocal8Bit();
-            qFatal( "%s", errorMsg.constData() );
+            LOG_WARNING("Unable to register dbus object for service" << servicePath);
```

Comment:
Code LGTM, maybe in the error messages we could also include: .. << servicePath << "terminating..."
Or similar, just to state in the logs that this is a deliberated shut-down due to the above reason.